internal/test/wsu/wsu_test.go

```diff
@@ -397,19 +415,52 @@ func testEastWestNetworking(t *testing.T, node *v1.Node, vm e2ef.WindowsVM) {
 
 	// test Windows <-> Windows on same node
-	// This will continually try to read from the Windows Server. We have to try multiple times as the Windows container
-	// takes some time to finish initial network setup.
-	winCurlerCommand := []string{"powershell.exe", "-command", "for (($i =0), ($j = 0); $i -lt 10; $i++) { " +
-		"$response = Invoke-Webrequest -UseBasicParsing -Uri " + winServerIP +
-		"; $code = $response.StatusCode; echo \"GET returned code $code\";" +
-		"If ($code -eq 200) {exit 0}; Start-Sleep -s 10;}; exit 1" + winServerIP}
+	winCurlerCommand := getWinCurlerCommand(winServerIP)
 	winCurlerJob, err := createWindowsServerJob("win-curler-"+vm.GetCredentials().GetInstanceId(), winCurlerCommand)
 	require.NoError(t, err, "Could not create Windows job")
 	defer deleteJob(winCurlerJob.Name)
 	err = waitUntilJobSucceeds(winCurlerJob.Name)
 	assert.NoError(t, err, "Could not curl the Windows webserver pod from a separate Windows container")
+}
+
+// testEastWestNetworkingAcrossVMs deploys Windows pods on two different VMs, and tests that the pods can communicate
+func testEastWestNetworkingAcrossVMs(t *testing.T, firstVM e2ef.WindowsVM, secondVM e2ef.WindowsVM) {
+	firstNode, err := framework.GetNode(firstVM.GetCredentials().GetIPAddress())
+	require.NoError(t, err, "Could not get Windows node object from first VM")
+
+	affinityForFirstNode, err := getAffinityForNode(firstNode)
+	require.NoError(t, err, "Could not get affinity for first node")
 
-	// TODO: test Windows <-> Windows on different node
+	// Deploy a webserver pod on the first node
+	winServerDeploymentOnFirstVM, err := deployWindowsWebServer("win-webserver-"+firstVM.GetCredentials().GetInstanceId(),
+		firstVM, affinityForFirstNode)
+	require.NoError(t, err, "Could not create Windows Server deployment on first VM")
+	defer deleteDeployment(winServerDeploymentOnFirstVM.Name)
+
+	// Get the pod so we can use its IP
+	winServerIP, err := getPodIP(*winServerDeploymentOnFirstVM.Spec.Selector)
+	require.NoError(t, err, "Could not retrieve pod with selector %v", *winServerDeploymentOnFirstVM.Spec.Selector)
+
+	// test Windows <-> Windows across nodes
+	winCurlerCommand := getWinCurlerCommand(winServerIP)
+	winCurlerJobOnSecondVM, err := createWindowsServerJob("win-curler-"+secondVM.GetCredentials().GetInstanceId(),
+		winCurlerCommand)
+	require.NoError(t, err, "Could not create Windows job on second VM")
+	defer deleteJob(winCurlerJobOnSecondVM.Name)
+	err = waitUntilJobSucceeds(winCurlerJobOnSecondVM.Name)
+	assert.NoError(t, err, "Could not curl the Windows webserver pod on the first VM from Windows container "+
+		"on the second VM")
+}
+
+// getWinCurlerCommand generates a command to curl a Windows server from the given IP address
+func getWinCurlerCommand(winServerIP string) []string {
+	// This will continually try to read from the Windows Server. We have to try multiple times as the Windows container
+	// takes some time to finish initial network setup.
```

Comment:
Is there a more programmatic way to detect when the initial network setup is complete? Otherwise this test can be flaky. I realize this was existing code but given we are adding more tests around this I am hoping we don't introduce more flakes.

Reply:
As per the discussions, this will be handled as part of a separate ticket